pkg/kn/commands/human_readable_flags.go

```diff
@@ -42,15 +42,20 @@ func (f *HumanPrintFlags) AllowedFormats() []string {
 // ToPrinter receives returns a printer capable of
 // handling human-readable output.
 func (f *HumanPrintFlags) ToPrinter(getHandlerFunc func(h hprinters.PrintHandler)) (hprinters.ResourcePrinter, error) {
-	p := hprinters.NewTablePrinter(hprinters.PrintOptions{})
+	p := hprinters.NewTablePrinter(hprinters.PrintOptions{f.AllNamespaces})
 	getHandlerFunc(p)
 	return p, nil
 }
 
 // AddFlags receives a *cobra.Command reference and binds
 // flags related to human-readable printing to it
 func (f *HumanPrintFlags) AddFlags(c *cobra.Command) {
-	//TODO: Add more flags as required
+	// Bind AllNamespaces flag
+	all, err := c.Flags().GetBool("all-namespaces")
+	if err != nil {
+		fmt.Fprintln(c.OutOrStdout(), err)
+	}
+	f.AllNamespaces = all
```

Comment:
Can we do make this other way round and handle it in more generic way ?

As in, flags defined in HumanPrintFlags, gets added to all resource listing commands.

ToPrinter handlers it for services, revisions and routes.

WDYT ?

Reply:
I'm not sure I can handle it in more generic way, but I changed to handle `--all-namespace` option in a similar way to [kubectl get](https://github.com/kubernetes/kubernetes/blob/2ca023f270779a7f017cf10b0e56d008262ed577/pkg/kubectl/cmd/get/humanreadable_flags.go#L30).